gnat2goto/driver/tree_walk.adb

```diff
@@ -880,6 +894,97 @@ package body Tree_Walk is
       return Do_Bare_Range_Constraint (N, Underlying);
    end Do_Array_Range;
 
+   ------------------
+   -- Add_To_Table --
+   ------------------
+
+   function Add_To_Table (Name : String; Symbol_Type : Irep; Value : Irep)
+                           return Symbol is
+      New_Symbol : Symbol;
+   begin
+      New_Symbol.SymType := Symbol_Type;
+      New_Symbol.Name := Intern (Name);
+      New_Symbol.PrettyName := New_Symbol.Name;
+      New_Symbol.BaseName := New_Symbol.Name;
+      New_Symbol.Mode := Intern ("C");
+      New_Symbol.Value := Value;
+
+      if not (Global_Symbol_Table.Contains (Key => Intern (Name))) then
+         Global_Symbol_Table.Insert (Intern (Name), New_Symbol);
+      end if;
+      return New_Symbol;
+   end Add_To_Table;
+
+   function Make_Range_Assert_Expr (N : Node_Id; Value : Irep;
+                                                 Value_Type : Irep)
+                                    return Irep
+   is
```

Comment:
This could use a comment showing (in pseudocode) what kind of code is being generated here

Reply:
Done.